models/demos/ttnn_resnet/tt/ttnn_functional_resnet50_new_conv_api.py

```diff
@@ -192,14 +191,37 @@ def run_downsample_if_req(
                     enable_split_reader=enable_split_reader,
                     enable_subblock_padding=enable_subblock_padding,
                 ),
+            }
+
+            if not ttnn.is_tensor_storage_on_device(self.ds_conv_weight_tensor):
+                self.ds_conv_weight_tensor = ttnn.prepare_conv_weights(
+                    weight_tensor=self.ds_conv_weight_tensor,
+                    weights_format="OIHW",
+                    input_memory_config=x.memory_config(),
+                    input_layout=x.get_layout(),
+                    **conv_kwargs,
+                )
+
+                self.ds_conv_bias_tensor = ttnn.prepare_conv_bias(
+                    bias_tensor=self.ds_conv_bias_tensor,
+                    input_memory_config=x.memory_config(),
+                    input_layout=x.get_layout(),
+                    **conv_kwargs,
+                )
+                self.ds_conv_weight_tensor = ttnn.to_device(self.ds_conv_weight_tensor, device)
```

Comment:
since weights and bias have to be explicitly moved to device, 
is it a better pattern to move them to device in init phase, and than just call prepare in runtime.